chef/cookbooks/bind9/recipes/default.rb

```diff
@@ -247,8 +245,8 @@ def make_zone(zone)
   next unless network.key?("allocated_by_name")
   net_name=network[:id].gsub(/_network$/, "").gsub("_","-")
   network[:allocated_by_name].each_key do |host|
-    if search(:node, "fqdn:#{host}").size > 0 or not host.match(/.#{node[:dns][:domain]}$/)
-      #this is node in crowbar terms or it not belong to our domain, so lets skip it
+    if !host.match(/.#{node[:dns][:domain]}$/) || fqdns.include?(host)
```

Comment:
I'm just moving the test which is less expensive first (it matters a bit when we have 100s of nodes).